src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -308,69 +317,103 @@ internal async Task ProcessMessage(Message receivedMessage, CancellationToken me
             if (!IsMessageExpired(receivedMessage, transportMessage.Headers, messageId, CorrectClockSkew.GetClockCorrectionForEndpoint(awsEndpointUrl)))
             {
                 // here we also want to use the native message id because the core demands it like that
-                await ProcessMessageWithInMemoryRetries(transportMessage.Headers, nativeMessageId, messageBody, receivedMessage, messageProcessingCancellationToken).ConfigureAwait(false);
+                await InnerProcessMessage(transportMessage.Headers, nativeMessageId, messageBody, transportMessage.S3BodyKey, receivedMessage, messageProcessingCancellationToken).ConfigureAwait(false);
```

Comment:
We can use the `bool` I proposed to return to check if we should call `DeleteMessage` or not